app/src/main/java/ru/alexzdns/fundamentals/homework/MainActivity.kt

```diff
@@ -1,20 +1,45 @@
 package ru.alexzdns.fundamentals.homework
 
-import android.content.Intent
 import android.os.Bundle
-import android.widget.Button
 import androidx.appcompat.app.AppCompatActivity
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(),
+    MovieListFragment.MovieListClickListener,
+    MovieDetailsFragment.MovieDetailsClickListener {
+    private var movieListFragment : MovieListFragment? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        val button = findViewById<Button>(R.id.ma_button)
-
-        button.setOnClickListener {
-            val intent = Intent(this, MovieDetailsActivity::class.java)
-            startActivity(intent)
+        if (savedInstanceState == null) {
+            movieListFragment = MovieListFragment().apply { setListener(this@MainActivity)}
+            supportFragmentManager.beginTransaction()
+                .add(R.id.ma_main_container, movieListFragment!!, MOVIE_LIST_FRAGMENT)
+                .commit()
+        } else {
+            movieListFragment = supportFragmentManager.findFragmentByTag(MOVIE_LIST_FRAGMENT) as? MovieListFragment
         }
     }
-}
\ No newline at end of file
+
+    override fun openMovieDetailsFragment() {
+        supportFragmentManager.beginTransaction()
+            .addToBackStack(null)
+            .add(
+                R.id.ma_main_container,
+                MovieDetailsFragment().apply { setListener(this@MainActivity) })
+            .commit()
+    }
+
+    override fun removeMovieDetailsFragment() {
+        val lastFragment = supportFragmentManager.fragments.last()
+
+        supportFragmentManager.beginTransaction()
+            .remove(lastFragment)
+            .commit()
```

Comment:
Достаточно просто вытолкнуть элемент из стека.
```suggestion
        supportFragmentManager.popBackStack()
```